src/components/PopupWithForm.js

```diff
@@ -35,14 +39,29 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
+  _renderLoading(isLoading) {
+    this._submitButton.textContent = isLoading
+      ? "Saving..."
+      : this._submitButtonText;
+  }
+
   // handle outside click closure & form submission
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
-      this._popupForm.reset(); // reset from input after successful form submission
-      this.close();
+      // this._handleFormSubmit(this._getInputValues());
+      // this._popupForm.reset(); // reset from input after successful form submission
+      // this.close();
+
+      this._renderLoading(true); // show Saving...
+      this._handleFormSubmit(this._getInputValues())
+        .then(() => {
+          this.close();
+        })
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.


You can add it here not to duplicate in `index.js`
```js

      this._renderLoading(true); // show Saving...
      this._handleFormSubmit(this._getInputValues())
        .then(() => {
          this.close();
          this._popupForm.reset();  // reset the form in `then`
        })
        .catch(console.error)  // catch possible errors
        .finally(() => {
          this._renderLoading(false); // revert to original text
        });
```